Remove c indexing 4

agitator opened the pull request.

agitator commented:
This is not removing support for 4.2, but having a test setup running on 3 different plone version is hassle enough.
Check Travis, jobs are running for 4.3, 5.0 and 5.1
I also pushed the version to 2.0

hvelarde commented on .travis.yml:
```diff
@@ -1,23 +1,25 @@
+python:
+  - 2.7.13
 env:
-  - PLONE_VERSION=4.2
```
if you're not removing support for Plone 4.2 why did removed this line?

it's totally valid to remove support for Plone 4.3 but you need to add that to the changelog.

hvelarde commented on dexterity/membrane/membrane_helpers.py:
```diff
@@ -36,7 +37,7 @@ def get_brains_for_email(context, email, request=None):
     #     try:
     #         request = context.REQUEST
-    #     except:
+    #     except:  # noqa
```
this is a bad idea; you have to check for `AttributeError` in the exception.

hvelarde commented on setup.py:
```diff
@@ -36,10 +37,9 @@
     zip_safe=False,
     install_requires=[
-        'Products.CMFPlone>=4.2',
+        'Products.CMFPlone>=4.3',
```
you are dropping support for Plone 4.2 explicitly here.

hvelarde commented on .travis.yml:
```diff
@@ -1,23 +1,25 @@
+  - virtualenv .
+  - bin/pip install -r requirements.txt
+  - sed -ie "s#plone-5.1.x.cfg#plone-$PLONE_VERSION.x.cfg#" travis.cfg
+  - bin/buildout -c travis.cfg buildout:download-cache=downloads code-analysis:return-status-codes=True
```
please add a `bin/buildout annotate` above he so we can check that the right versions are being tested.

agitator commented:
Sorry if I was unclear...
I dropped support for 4.2, since I don't wanted to setup tests for 4.2, the new versions where hassle enough.
I added a 1.2.x branch for fixes

jensens commented:
+1 for a legacy branch and pushing master forward

hvelarde commented:
and that's exactly why fixing the builds, code analysis and so on should be done outside this PR, to leave the legacy branch on a cleaner state.
